### ISSUE TITLE
BBB 2.2.31 - Stuck with "Connecting to echo test ..." window to use microphone

### ISSUE THREAD

**Issue by lovingkindness** [bigbluebutton/greenlight#2382]
Hello,
Thank you very much for your effort and generosity to share BigBlueButton. We appreciate it very much.

After upgrading to BigBlueButton 2.2.31, BBB gets stuck with "Connecting to echo test ..." window by choosing to use microphone on Greenlight. We can not use BBB for now.

--

Environment: VPS 4vcpu 8GB, Ubuntu 16.04 LTS, BBB and Greenlight installed.

Here is the reproduction procedure on Firefox v84.0.

1. Sing in Greenlight
2. Choose the room
then
3. Click "Start"

4. How would you like to join the audio
          Microphone   Listen only

5. Choose "Microphone"

6. Connecting to echo test ...
 |      Will you allow xxx to use your microphone - 
 |      Choose "ALLOW"
6. Connecting to echo test ...

Then process completely stops with the "Connecting to echo test ..." window. The 3 dots"connectingAnimation--2clGX5" keeps running to proceed to next step. But nothing happens anymore no matter how long we wait.

The window of "Connecting to echo test ..." even can not be closed with close "X" upper right portion of the window.

I just have to close the browser tab itself to end the session.

--

In case of Chrome

Sing in Greenlight
Choose the room
then
Click "Start"

How would you like to join the audio?
Microphone  Listen only

Microphone

then immediately this window pops up.

There was an issue with your media devices
Did you give permission for access to your microphone? Note that a dialog should appear when you try to join audio, asking for your media device permissions, please accept that in order to join the audio conference. If that is not the case, try changing your microphone permissions in your browser's settings.
                                                    Back

--

After clicking the "Back"

How would you like to join the audio?
        Microphone  Listen only

Choose "Microphone"

Connecting to echo test ...

There was an issue with your media devices
Did you give permission for access to your microphone? Note that a dialog should appear when you try to join audio, asking for your media device permissions, please accept that in order to join the audio conference. If that is not the case, try changing your microphone permissions in your browser's settings.
                                                    Back

--

After clicking the "Back"

How would you like to join the audio?
       Microphone  Listen only

Choose "Microphone"

Connecting to echo test ...

There was an issue with your media devices
Did you give permission for access to your microphone? Note that a dialog should appear when you try to join audio, asking for your media device permissions, please accept that in order to join the audio conference. If that is not the case, try changing your microphone permissions in your browser's settings.
                                                    Back

--

Continuously it keeps looping above cycle.

--

For your information, here is the "apt-cache policy bbb*" before we upgraded.
Before the upgrading, BBB had been working very fine for us, usually 10-15 participants.

apt-cache policy bbb*
bbb-apps-video:
  Installed: 1:2.2.0-48
  Candidate: 1:2.2.0-49
  Version table:
     1:2.2.0-49 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
 *** 1:2.2.0-48 100
        100 /var/lib/dpkg/status
bbb-mkclean:
  Installed: 1:0.8.7-3
  Candidate: 1:0.8.7-3
  Version table:
 *** 1:0.8.7-3 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
        100 /var/lib/dpkg/status
bbb-transcode-akka:
  Installed: 2.2.0-8
  Candidate: 2.2.0-8
  Version table:
 *** 2.2.0-8 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main i386 Packages
        100 /var/lib/dpkg/status
bbb-apps:
  Installed: 1:2.2.0-50
  Candidate: 1:2.2.0-51
  Version table:
     1:2.2.0-51 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
 *** 1:2.2.0-50 100
        100 /var/lib/dpkg/status
bbb-playback-presentation:
  Installed: 1:2.2.0-33
  Candidate: 1:2.2.0-35
  Version table:
     1:2.2.0-35 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
 *** 1:2.2.0-33 100
        100 /var/lib/dpkg/status
bbb-playback-screenshare:
  Installed: (none)
  Candidate: 1:2.2.0-14
  Version table:
     1:2.2.0-14 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
bbb-apps-video-broadcast:
  Installed: 1:2.2.0-8
  Candidate: 1:2.2.0-8
  Version table:
 *** 1:2.2.0-8 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
        100 /var/lib/dpkg/status
bbb-lti:
  Installed: (none)
  Candidate: 1:2.2.0-33
  Version table:
     1:2.2.0-33 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
bbb-check:
  Installed: (none)
  Candidate: 1:2.2.0-5
  Version table:
     1:2.2.0-5 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
bbb-apps-sip:
  Installed: 1:2.2.0-11
  Candidate: 1:2.2.0-11
  Version table:
 *** 1:2.2.0-11 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
        100 /var/lib/dpkg/status
bbb-swftools:
  Installed: (none)
  Candidate: 1:0.9.2-2
  Version table:
     1:0.9.2-2 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
bbb-html5:
  Installed: 1:2.2.0-1095
  Candidate: 1:2.2.0-1114
  Version table:
     1:2.2.0-1114 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
 *** 1:2.2.0-1095 100
        100 /var/lib/dpkg/status
bbb-webhooks:
  Installed: (none)
  Candidate: 1:2.2.0-19
  Version table:
     1:2.2.0-19 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
bbb-freeswitch-sounds:
  Installed: 1:1.6.7-6
  Candidate: 1:1.6.7-6
  Version table:
 *** 1:1.6.7-6 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
        100 /var/lib/dpkg/status
bbb-record-core:
  Installed: 1:2.2.0-71
  Candidate: 1:2.2.0-72
  Version table:
     1:2.2.0-72 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
 *** 1:2.2.0-71 100
        100 /var/lib/dpkg/status
bbb-etherpad:
  Installed: 1:2.2.0-63
  Candidate: 1:2.2.0-66
  Version table:
     1:2.2.0-66 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
 *** 1:2.2.0-63 100
        100 /var/lib/dpkg/status
bbb-web:
  Installed: 1:2.2.0-148
  Candidate: 1:2.2.0-150
  Version table:
     1:2.2.0-150 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
 *** 1:2.2.0-148 100
        100 /var/lib/dpkg/status
bbb-fsesl-akka:
  Installed: 2.2.0-72
  Candidate: 2.2.0-73
  Version table:
     2.2.0-73 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main i386 Packages
 *** 2.2.0-72 100
        100 /var/lib/dpkg/status
bbb-apps-screenshare:
  Installed: 1:2.2.0-50
  Candidate: 1:2.2.0-51
  Version table:
     1:2.2.0-51 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
 *** 1:2.2.0-50 100
        100 /var/lib/dpkg/status
bbb-freeswitch-core:
  Installed: 2:2.2.0-113
  Candidate: 2:2.2.0-113
  Version table:
 *** 2:2.2.0-113 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
        100 /var/lib/dpkg/status
bbb-playback-podcast:
  Installed: (none)
  Candidate: 1:2.2.0-6
  Version table:
     1:2.2.0-6 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
bbb-apps-akka:
  Installed: 2.2.0-96
  Candidate: 2.2.0-97
  Version table:
     2.2.0-97 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main i386 Packages
 *** 2.2.0-96 100
        100 /var/lib/dpkg/status
bbb-red5:
  Installed: 1:1.0.10-16
  Candidate: 1:1.0.10-16
  Version table:
 *** 1:1.0.10-16 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
        100 /var/lib/dpkg/status
bbb-client:
  Installed: 1:2.2.0-40
  Candidate: 1:2.2.0-40
  Version table:
 *** 1:2.2.0-40 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
        100 /var/lib/dpkg/status
bbb-config:
  Installed: 1:2.2.0-225
  Candidate: 1:2.2.0-234
  Version table:
     1:2.2.0-234 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
 *** 1:2.2.0-225 100
        100 /var/lib/dpkg/status
bbb-demo:
  Installed: (none)
  Candidate: 1:2.2.0-9
  Version table:
     1:2.2.0-9 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
bbb-playback-notes:
  Installed: (none)
  Candidate: 1:2.2.0-6
  Version table:
     1:2.2.0-6 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
bbb-webrtc-sfu:
  Installed: 1:2.2.0-108
  Candidate: 1:2.2.0-110
  Version table:
     1:2.2.0-110 500
        500 https://ubuntu.bigbluebutton.org/xenial-22 bigbluebutton-xenial/main amd64 Packages
 *** 1:2.2.0-108 100
        100 /var/lib/dpkg/status

--

After upgrading,  "dpkg -l | grep bbb*"

ii  bbb-apps                              1:2.2.0-51                                    amd64        BigBlueButton applications for Red5
ii  bbb-apps-akka                         2.2.0-97                                      all          BigBlueButton Apps (Akka)
ii  bbb-apps-screenshare                  1:2.2.0-51                                    amd64        BigBlueButton screenshare module for Red5
ii  bbb-apps-sip                          1:2.2.0-11                                    amd64        BigBlueButton SIP module for Red5
ii  bbb-apps-video                        1:2.2.0-49                                    amd64        BigBlueButton video module for Red5
ii  bbb-apps-video-broadcast              1:2.2.0-8                                     amd64        BigBlueButton video broadcast module for Red5
ii  bbb-client                            1:2.2.0-40                                    amd64        BigBlueButton Flash client
ii  bbb-config                            1:2.2.0-234                                   amd64        BigBlueButton configuration utilities
ii  bbb-etherpad                          1:2.2.0-66                                    amd64        The EtherPad Lite components for BigBlueButton
ii  bbb-freeswitch-core                   2:2.2.0-113                                   amd64        BigBlueButton build of FreeSWITCH
ii  bbb-freeswitch-sounds                 1:1.6.7-6                                     amd64        FreeSWITCH Sounds
ii  bbb-fsesl-akka                        2.2.0-73                                      all          BigBlueButton FS-ESL (Akka)
ii  bbb-html5                             1:2.2.0-1114                                  amd64        The HTML5 components for BigBlueButton
ii  bbb-mkclean                           1:0.8.7-3                                     amd64        Clean and optimize Matroska and WebM files
ii  bbb-playback-presentation             1:2.2.0-35                                    amd64        BigBluebutton playback of presentation
ii  bbb-record-core                       1:2.2.0-72                                    amd64        BigBlueButton record and playback
ii  bbb-red5                              1:1.0.10-16                                   amd64        The Red5 server for BigBlueButton
ii  bbb-transcode-akka                    2.2.0-8                                       all          BigBlueButton Transcoder
ii  bbb-web                               1:2.2.0-150                                   amd64        BigBlueButton API
ii  bbb-webrtc-sfu                        1:2.2.0-110                                   amd64        BigBlueButton WebRTC SFU
ii  bigbluebutton                         1:2.2.0-5                                     amd64        Open source web conferencing platform (bbb)
ii  ffmpeg                                7:4.2.4-1ubuntu0.1bbb1~16.04.1                amd64        Tools for transcoding, streaming and playing of multimedia files
ii  libavcodec58:amd64                    7:4.2.4-1ubuntu0.1bbb1~16.04.1                amd64        FFmpeg library with de/encoders for audio/video codecs - runtime files
ii  libavdevice58:amd64                   7:4.2.4-1ubuntu0.1bbb1~16.04.1                amd64        FFmpeg library for handling input and output devices - runtime files
ii  libavfilter7:amd64                    7:4.2.4-1ubuntu0.1bbb1~16.04.1                amd64        FFmpeg library containing media filters - runtime files
ii  libavformat58:amd64                   7:4.2.4-1ubuntu0.1bbb1~16.04.1                amd64        FFmpeg library with (de)muxers for multimedia containers - runtime files
ii  libavresample4:amd64                  7:4.2.4-1ubuntu0.1bbb1~16.04.1                amd64        FFmpeg compatibility library for resampling - runtime files
ii  libavutil56:amd64                     7:4.2.4-1ubuntu0.1bbb1~16.04.1                amd64        FFmpeg library with functions for simplifying programming - runtime files
ii  libblkid1:amd64                       2.27.1-6ubuntu3.10                            amd64        block device ID library
ii  libbluray1:amd64                      1:0.9.2-2                                     amd64        Blu-ray disc playback support library (shared library)
ii  libboost-date-time1.58.0:amd64        1.58.0+dfsg-5ubuntu3.1                        amd64        set of date-time libraries based on generic programming concepts
ii  libboost-filesystem1.58.0:amd64       1.58.0+dfsg-5ubuntu3.1                        amd64        filesystem operations (portable paths, iteration over directories, etc) in C++
ii  libboost-iostreams1.58.0:amd64        1.58.0+dfsg-5ubuntu3.1                        amd64        Boost.Iostreams Library
ii  libboost-locale1.58.0:amd64           1.58.0+dfsg-5ubuntu3.1                        amd64        C++ facilities for localization
ii  libboost-log1.58.0                    1.58.0+dfsg-5ubuntu3.1                        amd64        C++ logging library
ii  libboost-program-options1.58.0:amd64  1.58.0+dfsg-5ubuntu3.1                        amd64        program options library for C++
ii  libboost-regex1.58.0:amd64            1.58.0+dfsg-5ubuntu3.1                        amd64        regular expression library for C++
ii  libboost-system1.58.0:amd64           1.58.0+dfsg-5ubuntu3.1                        amd64        Operating system (e.g. diagnostics support) library
ii  libboost-thread1.58.0:amd64           1.58.0+dfsg-5ubuntu3.1                        amd64        portable C++ multi-threading
ii  libbs2b0:amd64                        3.1.0+dfsg-2.2                                amd64        Bauer stereophonic-to-binaural DSP library
ii  libbsd0:amd64                         0.8.2-1ubuntu0.1                              amd64        utility functions from BSD systems - shared library
ii  libbz2-1.0:amd64                      1.0.6-8ubuntu0.2                              amd64        high-quality block-sorting file compressor library - runtime
ii  libopusenc0                           0.2.1-1bbb1                                   amd64        High-level API for encoding Ogg Opus audio streams
ii  libpostproc55:amd64                   7:4.2.4-1ubuntu0.1bbb1~16.04.1                amd64        FFmpeg library for post processing - runtime files
ii  librubberband2v5:amd64                1.8.1-6ubuntu2                                amd64        audio time-stretching and pitch-shifting library
ii  libswresample3:amd64                  7:4.2.4-1ubuntu0.1bbb1~16.04.1                amd64        FFmpeg library for audio resampling, rematrixing etc. - runtime files
ii  libswscale5:amd64                     7:4.2.4-1ubuntu0.1bbb1~16.04.1                amd64        FFmpeg library for image scaling and various conversions - runtime files
ii  libtbb2:amd64                         4.4~20151115-0ubuntu3                         amd64        parallelism library for C++ - runtime files

---

Thank you,

May you, your family and all beings around you be well, happy and peaceful.

**Comment by Dennis1993** [comment]
Hey,

I‘ve the same problem. I fixed it by runnig the following Commander:
`bbb-conf —setip bbb.mydomain.de`

After that I typed in
`bbb-conf—clean`

Now it works. In this situation I‘ve upgraded Greenlight too with
`cd ~`
`cd greenlight/`
`docker-compose pull`
`docker-compose down`
`docker-compose up -d`

Now are all components are up to date and works fine 😊

I hope I can help you

**Comment by lovingkindness** [comment]
Hello, Dennis1993
Thank you very much for your help.

I followed your instruction.
Here are the commands I run.
All commands run without any problem.

--

bbb-conf --setip [hostname.domain]
bbb-conf --clean

cd ~
cd greenlight/
docker-compose pull
docker-compose down
docker-compose up -d

--

But unfortunately when I run Firefox and accessed https://hostname.domain, I get this page of NginX.


404 Not Found
nginx


--

Now I lost the Greenlight also. Hahaha.
Do you think I am going down further deeper into this swamp?
:-)

I also place the command info below.
Any advice will be appreciated.

Thank you

--


bbb-conf --setip [HOSTNAME.DOMAIN]
Assigning [HOSTNAME.DOMAIN] for testing for firewall in /var/www/bigbluebutton/client/conf/config.xml
Assigning [HOSTNAME.DOMAIN] for rtmp:// in /var/www/bigbluebutton/client/conf/config.xml
Assigning [HOSTNAME.DOMAIN] for servername in /etc/nginx/sites-available/bigbluebutton
Assigning [HOSTNAME.DOMAIN] for http[s]:// in /var/www/bigbluebutton/client/conf/config.xml
Assigning [HOSTNAME.DOMAIN] for publishURI in /var/www/bigbluebutton/client/conf/config.xml
Assigning [HOSTNAME.DOMAIN] for web application URL in /usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties
Assigning [HOSTNAME.DOMAIN] for web application URL in /usr/share/bbb-apps-akka/conf/application.conf
Assigning [HOSTNAME.DOMAIN] for record and playback in /usr/local/bigbluebutton/core/scripts/bigbluebutton.yml
Assigning [HOSTNAME.DOMAIN] for playback of recordings:
Restarting the BigBlueButton 2.2.31 ...
Stopping BigBlueButton


Applying updates in /etc/bigbluebutton/bbb-conf/apply-config.sh:
  - Enable Firewall and opening 22/tcp, 80/tcp, 443/tcp and 16384:32768/udp
Skipping adding existing rule
Skipping adding existing rule (v6)
Skipping adding existing rule
Skipping adding existing rule (v6)
Skipping adding existing rule
Skipping adding existing rule (v6)
Skipping adding existing rule
Skipping adding existing rule (v6)
Firewall is active and enabled on system startup

Starting BigBlueButton


bbb-conf --clean
Restarting BigBlueButton 2.2.31 (and cleaning out all log files) ...
Stopping BigBlueButton
 ... cleaning log files

Applying updates in /etc/bigbluebutton/bbb-conf/apply-config.sh:
  - Enable Firewall and opening 22/tcp, 80/tcp, 443/tcp and 16384:32768/udp
Skipping adding existing rule
Skipping adding existing rule (v6)
Skipping adding existing rule
Skipping adding existing rule (v6)
Skipping adding existing rule
Skipping adding existing rule (v6)
Skipping adding existing rule
Skipping adding existing rule (v6)
Firewall is active and enabled on system startup

Starting BigBlueButton
................


Potential problems described below
      Log file too short (less than 100 lines): /usr/share/red5/log/bigbluebutton.log (red5)

..........

cd ~
ls -al
total 64
drwx------  9 root root 4096 Dec  3 22:27 .
drwxr-xr-x 22 root root 4096 Dec 19 15:49 ..
-rw-------  1 root root 2979 Dec 18 17:37 .bash_history
-rw-r--r--  1 root root 3106 Oct 22  2015 .bashrc
drwxr-xr-x  3 root root 4096 Dec  3 20:36 .bundle
drwx------  3 root root 4096 Dec  3 20:36 .cache
-rw-r--r--  1 root root    0 Nov  1 05:31 .cloud-locale-test.skip
drwx------  4 root root 4096 Dec  3 20:36 .config
-rw-------  1 root root   22 Dec 19 17:32 .fs_cli_history
drwxr-xr-x  3 root root 4096 Dec  3 20:36 .gem
drwxr-xr-x  5 root root 4096 Dec  3 21:58 greenlight
-rw-------  1 root root   28 Dec  3 20:16 .lesshst
drwxr-xr-x  2 root root 4096 Dec  3 21:07 .nano
-rw-------  1 root root   17 Dec 17 20:47 .npmrc
-rw-r--r--  1 root root  148 Aug 17  2015 .profile
-rw-------  1 root root 1024 Dec 19 17:31 .rnd
drwx------  2 root root 4096 Dec  3 11:35 .ssh

cd greenlight
~/greenlight# ls -al
total 48
drwxr-xr-x 5 root root  4096 Dec  3 21:58 .
drwx------ 9 root root  4096 Dec  3 22:27 ..
drwxr-xr-x 3 root root  4096 Dec  3 20:40 db
-rw-r--r-- 1 root root   877 Dec  3 20:40 docker-compose.yml
-rw-r--r-- 1 root root 10643 Dec  3 20:40 .env
-rw-r--r-- 1 root root 10643 Dec  3 21:50 .env_backup
drwxr-xr-x 2 root root  4096 Dec  3 20:40 log
drwxr-xr-x 2 root root  4096 Dec  3 20:40 storage

~/greenlight# docker-compose pull
Pulling db  ... done
Pulling app ... done

~/greenlight# docker-compose down
Stopping greenlight-v2   ... done
Stopping greenlight_db_1 ... done
Removing greenlight-v2   ... done
Removing greenlight_db_1 ... done
Removing network greenlight_default

~/greenlight# docker-compose up -d
Creating network "greenlight_default" with the default driver
Creating greenlight_db_1 ... done
Creating greenlight-v2   ... done

~/greenlight#


--

bbb-conf --check

BigBlueButton Server 2.2.31 (2307)
                    Kernel version: 4.4.0-197-generic
                      Distribution: Ubuntu 16.04.7 LTS (64-bit)
                            Memory: 7980 MB
                         CPU cores: 4

/usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties (bbb-web)
       bigbluebutton.web.serverURL: https://[HOSTNAME.DOMAIN]
                defaultGuestPolicy: ALWAYS_ACCEPT
                 svgImagesRequired: true

/etc/nginx/sites-available/bigbluebutton (nginx)
                       server_name: [HOSTNAME.DOMAIN]
                              port: 80, [::]:80
                              port: 443 ssl

/opt/freeswitch/etc/freeswitch/vars.xml (FreeSWITCH)
                       local_ip_v4: xxx.xxx.xxx.xxx
                   external_rtp_ip: xxx.xxx.xxx.xxx
                   external_sip_ip: xxx.xxx.xxx.xxx

/opt/freeswitch/etc/freeswitch/sip_profiles/external.xml (FreeSWITCH)
                        ext-rtp-ip: $${local_ip_v4}
                        ext-sip-ip: $${local_ip_v4}
                        ws-binding: :5066
                       wss-binding: xxx.xxx.xxx.xxx:7443

/usr/local/bigbluebutton/core/scripts/bigbluebutton.yml (record and playback)
                     playback_host: [HOSTNAME.DOMAIN]
                 playback_protocol: https
                            ffmpeg: 4.2.4-1ubuntu0.1bbb1~16.04.1

/etc/bigbluebutton/nginx/sip.nginx (sip.nginx)
                        proxy_pass: xxx.xxx.xxx.xxx
                          protocol: https

/usr/local/bigbluebutton/bbb-webrtc-sfu/config/default.yml (Kurento SFU)
                        kurento.ip: xxx.xxx.xxx.xxx
                       kurento.url: ws://127.0.0.1:8888/kurento
                    kurento.sip_ip: xxx.xxx.xxx.xxx
                    localIpAddress: xxx.xxx.xxx.xxx
               recordScreenSharing: true
                     recordWebcams: true
                  codec_video_main: VP8
               codec_video_content: VP8

/usr/share/meteor/bundle/programs/server/assets/app/config/settings.yml (HTML5 client)
                             build: 1114
                        kurentoUrl: wss://[HOSTNAME.DOMAIN]/bbb-webrtc-sfu
                  enableListenOnly: true
                    sipjsHackViaWs: false

/usr/share/bbb-web/WEB-INF/classes/spring/turn-stun-servers.xml (STUN Server)
                              stun:[HOSTNAME2.DOMAIN]


Potential problems described below



--

less /usr/share/red5/log/bigbluebutton.log
2020-12-19T17:32:40.941+01:00 INFO  ROOT - Starting up context: bigbluebutton
2020-12-19T17:32:52.705+01:00 DEBUG o.b.red5.BigBlueButtonApplication - start: Scope [name=bigbluebutton, path=/default, type=APPLICATION, autoStart=true, creationTime=1608395572687, depth=1, enabled=true, running=false]
2020-12-19T17:32:52.760+01:00 DEBUG o.b.red5.BigBlueButtonApplication - appStart: Scope [name=bigbluebutton, path=/default, type=APPLICATION, autoStart=true, creationTime=1608395572687, depth=1, enabled=true, running=false]
2020-12-19T17:32:52.946+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":483,"max":483,"used":63,"free":419}
2020-12-19T17:32:56.220+01:00 INFO  o.b.c.redis.pubsub.MessageSender - Redis org.bigbluebutton.red5.pubsub.message publisher starting!
2020-12-19T17:32:57.260+01:00 INFO  o.b.c.redis.pubsub.MessageSender - Redis org.bigbluebutton.red5.pubsub.message publisher starting!
2020-12-19T17:32:57.761+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":484,"max":484,"used":53,"free":430}
2020-12-19T17:33:02.761+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":484,"max":484,"used":89,"free":394}
2020-12-19T17:33:07.761+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":483,"max":483,"used":79,"free":403}
2020-12-19T17:33:12.762+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":483,"max":483,"used":81,"free":401}
2020-12-19T17:33:17.761+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":483,"max":483,"used":93,"free":389}
2020-12-19T17:33:22.761+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":483,"max":483,"used":93,"free":389}
2020-12-19T17:33:27.761+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":483,"max":483,"used":105,"free":377}
2020-12-19T17:33:32.761+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":483,"max":483,"used":106,"free":376}
2020-12-19T17:33:37.761+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":483,"max":483,"used":118,"free":364}
2020-12-19T17:33:42.761+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":483,"max":483,"used":118,"free":364}
2020-12-19T17:33:47.761+01:00 INFO  o.b.red5.BigBlueButtonApplication - JVM Heap [MB] data={"total":483,"max":483,"used":125,":

--


Thank you and have a nice weekend.

May you, your family and all beings around you be well, happy and peaceful.

**Comment by Dennis1993** [comment]
Oh thats bad... same problem after Server-restart?

Have you wait a minute? If I restart docker the website needs 1 minute. Before that the error 404 is shown like your described report.

if not: Whats the output of `dockerlogs greenlight-v2` and `docker logs greenlight_db_1`?

**Comment by lovingkindness** [comment]
Hi Dennis1993,
Here is an update.

Before I shutdown the server for today, I had rebooted it and tested with Firefox to see if I still get 404 error.

After rebooting the server, the site appeared normally!
I could Sign in. But microphone still do not work. But the symptom is a little different.

I describe it below.


--

Sign in to Greenlight

Choose the room

Click "Start"

How would you like to join the audio
      Microphone   Listen only

--

Choose "Microphone"

Connecting to echo
  Then small separate window pops up with this message.
        |Failer on call (Reason=ICE error)
        (error 1004)
  Then automatically "Connecting" window switches to "How" window.
How would you like to join the audio
      Microphone   Listen only

--

Choose "Microphone"

Connecting to echo
  Then small separate window pops up with this message.
        |Failer on call (Reason=ICE error)
        (error 1004)
  Then automatically "Connecting" window switches to "How" window.
How would you like to join the audio
      Microphone   Listen only

--

Above loop continues.


If I choose "Listen only" instead of "Microphone", initially the icon of "Listen only" is highlighted with blue circle but after a few seconds the blue goes away and nothing happens. The window just stays on the screen. But this time, "How would you like to join the audio" window can be closed with close X. So I can join the session.

Although "Join Audio" does not work, "Share webcam" function works normally.

So obviously there is an improvement. Thank you.


Oh, I just received your message. Thank you.
Obviously I did not wait for 1 min after restarting docker. So it makes sense that I got error 404 as you mentioned.
I add the results of "docker logs greenlight-v2" and "docker logs greenlight_db_1" below for your reference.
Thank you,


--

docker logs greenlight-v2
curl: (52) Empty reply from server
Database 'greenlight_production' already exists
>>> Database migration
== 20201214232153 AddLastLoginToUsers: migrating ==============================
-- add_column(:users, :last_login, :datetime)
   -> 0.0046s
== 20201214232153 AddLastLoginToUsers: migrated (0.0047s) =====================

yarn install v1.3.2
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
Done in 0.07s.
I, [2020-12-19T16:34:38.635400 #20]  INFO -- : Writing /usr/src/app/public/assets/pickr.min-46a1bbeb0f07af31f94a0b5a3e2785681316f818d22d498ab8c00eb9a534b718.js
I, [2020-12-19T16:34:38.636277 #20]  INFO -- : Writing /usr/src/app/public/assets/pickr.min-46a1bbeb0f07af31f94a0b5a3e2785681316f818d22d498ab8c00eb9a534b718.js.gz
I, [2020-12-19T16:34:41.295994 #20]  INFO -- : Writing /usr/src/app/public/assets/monolith.min-34d021658b61d0d33e86880ebcdbeb1a9b705a2004cc564b834586d44ed23496.css
I, [2020-12-19T16:34:41.297099 #20]  INFO -- : Writing /usr/src/app/public/assets/monolith.min-34d021658b61d0d33e86880ebcdbeb1a9b705a2004cc564b834586d44ed23496.css.gz
I, [2020-12-19T16:34:42.903690 #20]  INFO -- : Writing /usr/src/app/public/assets/bootstrap-select.min-26d1d44da4b2446ef05c95a949d58f0b0e25a3f222875304f7f078f38962f7ea.js
I, [2020-12-19T16:34:42.903984 #20]  INFO -- : Writing /usr/src/app/public/assets/bootstrap-select.min-26d1d44da4b2446ef05c95a949d58f0b0e25a3f222875304f7f078f38962f7ea.js.gz
I, [2020-12-19T16:34:44.165853 #20]  INFO -- : Writing /usr/src/app/public/assets/bootstrap-select.min-ee9a5ed5e50fa4db59d026cf5a342f5d116f2f5afa789935ad86c24111935ce0.css
I, [2020-12-19T16:34:44.166207 #20]  INFO -- : Writing /usr/src/app/public/assets/bootstrap-select.min-ee9a5ed5e50fa4db59d026cf5a342f5d116f2f5afa789935ad86c24111935ce0.css.gz
I, [2020-12-19T16:34:44.173143 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-brands-400-931d173390b901b84b133796a2e42e54b82a4d60fc7f0baa23e05742af9793e8.eot
I, [2020-12-19T16:34:44.173671 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-brands-400-931d173390b901b84b133796a2e42e54b82a4d60fc7f0baa23e05742af9793e8.eot.gz
I, [2020-12-19T16:34:44.179603 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-regular-400-da05e5ee7c75c965efb151a6e87ab5589b8023f246f698c1d14946414bc31c90.eot
I, [2020-12-19T16:34:44.180162 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-regular-400-da05e5ee7c75c965efb151a6e87ab5589b8023f246f698c1d14946414bc31c90.eot.gz
I, [2020-12-19T16:34:44.188229 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-solid-900-d5342ae679d7064d475403ce5fe05d80f1a0abe5881f729ee0e3272a9c355f3b.eot
I, [2020-12-19T16:34:44.188855 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-solid-900-d5342ae679d7064d475403ce5fe05d80f1a0abe5881f729ee0e3272a9c355f3b.eot.gz
I, [2020-12-19T16:34:44.200892 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-brands-400-85eee11562ca8330dd6e3e7b726309ab95110e062d72c759abc83f897b8ed074.svg
I, [2020-12-19T16:34:44.202147 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-brands-400-85eee11562ca8330dd6e3e7b726309ab95110e062d72c759abc83f897b8ed074.svg.gz
I, [2020-12-19T16:34:44.209832 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-regular-400-e14de0404bcbe28ac413bd4c992f552af250da30823e635ab9a7656e50e77bc8.svg
I, [2020-12-19T16:34:44.210271 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-regular-400-e14de0404bcbe28ac413bd4c992f552af250da30823e635ab9a7656e50e77bc8.svg.gz
I, [2020-12-19T16:34:44.222327 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-solid-900-4ee1b170ff51b7fda11acef551d1ef492c0ebc1aff8e5712d3e30fcbb835776d.svg
I, [2020-12-19T16:34:44.223809 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-solid-900-4ee1b170ff51b7fda11acef551d1ef492c0ebc1aff8e5712d3e30fcbb835776d.svg.gz
I, [2020-12-19T16:34:44.229214 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-brands-400-d7746d2966e30323266a9f7ea24be80940f7a0207a0b91dde1669f3b3bdecd33.ttf
I, [2020-12-19T16:34:44.229581 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-brands-400-d7746d2966e30323266a9f7ea24be80940f7a0207a0b91dde1669f3b3bdecd33.ttf.gz
I, [2020-12-19T16:34:44.234981 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-regular-400-4e2fc1d3828b93e7e943c8017a542b0ad00f6333d28748658bb0697de23fd52e.ttf
I, [2020-12-19T16:34:44.235507 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-regular-400-4e2fc1d3828b93e7e943c8017a542b0ad00f6333d28748658bb0697de23fd52e.ttf.gz
I, [2020-12-19T16:34:44.245122 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-solid-900-9313de7304bef452ff5ac93e0f9ca48c3971d2fbb502c3fa1e2c2ea1f91ad853.ttf
I, [2020-12-19T16:34:44.246093 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-solid-900-9313de7304bef452ff5ac93e0f9ca48c3971d2fbb502c3fa1e2c2ea1f91ad853.ttf.gz
I, [2020-12-19T16:34:44.252155 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-brands-400-ef7234f1e9d303d2e42a209c547c217b90afa49d60502d04f1b14c906efc02d1.woff
I, [2020-12-19T16:34:44.258163 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-regular-400-684157e40f7ee2429707a23d5b77d2c71c5d23f26a9d82839275ff984d13053b.woff
I, [2020-12-19T16:34:44.262814 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-solid-900-c1450cddfd98668e6e593e2d26f11b4c43faaccb5f033b0b50ea6f1ce7bf4b58.woff
I, [2020-12-19T16:34:44.268452 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-brands-400-1f49b8706547682e2c5ed6642a2f2dcbd287da458314b967c60d774aa7edb473.woff2
I, [2020-12-19T16:34:44.272749 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-regular-400-6a16c04229bc2b4da226eb97e68d94f49ba6437b7b5e16c14a101b21a29384e9.woff2
I, [2020-12-19T16:34:44.278487 #20]  INFO -- : Writing /usr/src/app/public/assets/font-awesome/fa-solid-900-cd398be1a91817126cef10224738e624358edf6f08043abad7e60c1aaeccc8d0.woff2
I, [2020-12-19T16:34:44.285470 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/OFL-8ce06420e0277f1b28f514473263688414965ffdc13c3de4f1f0547a6d45f10b.txt
I, [2020-12-19T16:34:44.285790 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/OFL-8ce06420e0277f1b28f514473263688414965ffdc13c3de4f1f0547a6d45f10b.txt.gz
I, [2020-12-19T16:34:44.292211 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-Bold-9cbab47276fc04c65ac78098e9a2069c55e26f21701b29092734ce4e830f80fb.ttf
I, [2020-12-19T16:34:44.294090 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-Bold-9cbab47276fc04c65ac78098e9a2069c55e26f21701b29092734ce4e830f80fb.ttf.gz
I, [2020-12-19T16:34:44.302017 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-BoldItalic-8d5dd2d79ea1b5ba53c711016e5dbb0b659f94a46a3f6aa3f6bfd04f3e6ffff6.ttf
I, [2020-12-19T16:34:44.303013 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-BoldItalic-8d5dd2d79ea1b5ba53c711016e5dbb0b659f94a46a3f6aa3f6bfd04f3e6ffff6.ttf.gz
I, [2020-12-19T16:34:44.331176 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-Italic-1b78bc1c8104348e13fcc043331926f818ce290556350b1a90a735fe038c73d4.ttf
I, [2020-12-19T16:34:44.331725 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-Italic-1b78bc1c8104348e13fcc043331926f818ce290556350b1a90a735fe038c73d4.ttf.gz
I, [2020-12-19T16:34:44.347878 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-Light-3eb3ad852d98c139c78064804524d0d2472b3efb872dc176b217fb57ae6a6f78.ttf
I, [2020-12-19T16:34:44.353030 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-Light-3eb3ad852d98c139c78064804524d0d2472b3efb872dc176b217fb57ae6a6f78.ttf.gz
I, [2020-12-19T16:34:44.359724 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-LightItalic-3d845629b0f9c122ba4ca398e766ade519062d9623e72404613a61dac9565b7c.ttf
I, [2020-12-19T16:34:44.366296 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-LightItalic-3d845629b0f9c122ba4ca398e766ade519062d9623e72404613a61dac9565b7c.ttf.gz
I, [2020-12-19T16:34:44.381329 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-Regular-c9868de61ff2bab0b5a3a6d01c4b76f299459f08c6ae2f2c0383b4f9f6bedbf3.ttf
I, [2020-12-19T16:34:44.382051 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-Regular-c9868de61ff2bab0b5a3a6d01c4b76f299459f08c6ae2f2c0383b4f9f6bedbf3.ttf.gz
I, [2020-12-19T16:34:44.392524 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-SemiBold-3ba5c382a7ee6a8831bdf90192addceabe6db4278a679e67fe7e9c0226b729cf.ttf
I, [2020-12-19T16:34:44.393724 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-SemiBold-3ba5c382a7ee6a8831bdf90192addceabe6db4278a679e67fe7e9c0226b729cf.ttf.gz
I, [2020-12-19T16:34:44.410990 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-SemiBoldItalic-ad51e3c788064fcfd2ba4662df5732df3b57788cf20c70c1f3e313a6ddec7f9b.ttf
I, [2020-12-19T16:34:44.411346 #20]  INFO -- : Writing /usr/src/app/public/assets/SourceSansPro/SourceSansPro-SemiBoldItalic-ad51e3c788064fcfd2ba4662df5732df3b57788cf20c70c1f3e313a6ddec7f9b.ttf.gz
I, [2020-12-19T16:34:44.433415 #20]  INFO -- : Writing /usr/src/app/public/assets/background-03437e13cc10ea689f7ac9751fb0adcb0647a48af5c0af47ed34ec15948f02a4.png
I, [2020-12-19T16:34:44.442538 #20]  INFO -- : Writing /usr/src/app/public/assets/bbb_logo-75f7f251cbcb96efaf115f7b0ae7413cf589bf92cca7acc487fb36b55e8eefda.png
I, [2020-12-19T16:34:44.450517 #20]  INFO -- : Writing /usr/src/app/public/assets/favicon-76c923a0c96c0e77a6ec9d6596b30d597bf270a94136c36aac0cfae8d647b83c.ico
I, [2020-12-19T16:34:44.450741 #20]  INFO -- : Writing /usr/src/app/public/assets/favicon-76c923a0c96c0e77a6ec9d6596b30d597bf270a94136c36aac0cfae8d647b83c.ico.gz
I, [2020-12-19T16:34:44.457176 #20]  INFO -- : Writing /usr/src/app/public/assets/google-logo-4ac2ad99a3ff6d181822fcf94b78b1cbe473ccef3d5e1924a21521b31b762f7e.png
I, [2020-12-19T16:34:44.463084 #20]  INFO -- : Writing /usr/src/app/public/assets/landing/bbb-9f937740e56ec1efe62b55750e6f9e889ba8ba4d31ad466150dbdf0cc6c7638b.png
I, [2020-12-19T16:34:44.467731 #20]  INFO -- : Writing /usr/src/app/public/assets/landing/mobile-43e3539134ba531b30159d09a1e5d93ef601b8e1d10e1d06d0f9501d3220b5be.png
I, [2020-12-19T16:34:44.473656 #20]  INFO -- : Writing /usr/src/app/public/assets/landing/world-db81c084fd0f6b2a0d28ee482b6dca6d90a1771a2c436c6e591d3a3cd639ba10.png
I, [2020-12-19T16:34:44.479462 #20]  INFO -- : Writing /usr/src/app/public/assets/ldap-logo-640cfe5796ea8801aa0183dfd5446884b820c5256e991350cacec7f24457dc3a.png
I, [2020-12-19T16:34:44.484919 #20]  INFO -- : Writing /usr/src/app/public/assets/logo_with_text-92efcd53c4fd5da57bb688a5529aedb070114bc55902d45511386def234874fd.png
I, [2020-12-19T16:34:44.488470 #20]  INFO -- : Writing /usr/src/app/public/assets/office365-logo-02ff0eb61fa4d08b4c99ebf11eb421c41d8ec2f1c66a4a1f480762cad7834b26.jpg
I, [2020-12-19T16:34:44.494537 #20]  INFO -- : Writing /usr/src/app/public/assets/placeholder-user-ab6c04768c4e7f748d88e4028e10ed77423767fd1c9a1d8d59005191134c3df8.jpg
I, [2020-12-19T16:34:44.500293 #20]  INFO -- : Writing /usr/src/app/public/assets/twitter-logo-1b7eebbb6441be6d6ecb92539fa4a044247850c29e784bea9a298415d0414a23.png
I, [2020-12-19T16:34:44.505910 #20]  INFO -- : Writing /usr/src/app/public/assets/windows-logo-60621adc8d1d538d4269922ebe645d540d0591f190701cf2704143458965b928.png
I, [2020-12-19T16:35:03.010308 #20]  INFO -- : Writing /usr/src/app/public/assets/application-e68a8eba8d802f56a903c24916b6c298a96454fa3a38936a5be6e87f7811a5ad.js
I, [2020-12-19T16:35:03.010821 #20]  INFO -- : Writing /usr/src/app/public/assets/application-e68a8eba8d802f56a903c24916b6c298a96454fa3a38936a5be6e87f7811a5ad.js.gz
autoprefixer: /usr/src/app/app/assets/stylesheets/application.scss:13152:3: Replace text-decoration-skip: ink to text-decoration-skip-ink: auto, because spec had been changed
I, [2020-12-19T16:35:06.947292 #20]  INFO -- : Writing /usr/src/app/public/assets/application-0aeac8a042493a1f12c17011150489ebbcb7831f6e4c2f62d81e4d3a97266023.css
I, [2020-12-19T16:35:06.947520 #20]  INFO -- : Writing /usr/src/app/public/assets/application-0aeac8a042493a1f12c17011150489ebbcb7831f6e4c2f62d81e4d3a97266023.css.gz
[1] Puma starting in cluster mode...
[1] * Version 3.12.6 (ruby 2.5.1-p57), codename: Llamas in Pajamas
[1] * Min threads: 5, max threads: 5
[1] * Environment: production
[1] * Process workers: 1
[1] * Preloading application
[1] * Listening on tcp://0.0.0.0:80
[1] ! WARNING: Detected 1 Thread(s) started in app boot:
[1] ! #<Thread:0x000056430945ea28@/usr/src/app/vendor/bundle/ruby/2.5.0/gems/activerecord-5.2.4.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:299 sleep> - /usr/src/app/vendor/bundle/ruby/2.5.0/gems/activerecord-5.2.4.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:301:in `sleep'
[1] Use Ctrl-C to stop
[1] - Worker 0 (pid: 125) booted, phase: 0
[1] - Gracefully shutting down workers...
Waiting for postgres to start up ...
curl: (52) Empty reply from server
Database 'greenlight_production' already exists
>>> Database migration
yarn install v1.3.2
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.10s.
[1] Puma starting in cluster mode...
[1] * Version 3.12.6 (ruby 2.5.1-p57), codename: Llamas in Pajamas
[1] * Min threads: 5, max threads: 5
[1] * Environment: production
[1] * Process workers: 1
[1] * Preloading application
[1] * Listening on tcp://0.0.0.0:80
[1] ! WARNING: Detected 1 Thread(s) started in app boot:
[1] ! #<Thread:0x0000559a52414960@/usr/src/app/vendor/bundle/ruby/2.5.0/gems/activerecord-5.2.4.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:299 sleep> - /usr/src/app/vendor/bundle/ruby/2.5.0/gems/activerecord-5.2.4.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:301:in `sleep'
[1] Use Ctrl-C to stop
[1] - Worker 0 (pid: 52) booted, phase: 0


--


docker logs greenlight_db_1

PostgreSQL Database directory appears to contain a database; Skipping initialization

LOG:  database system was shut down at 2020-12-19 16:34:10 UTC
LOG:  MultiXact member wraparound protections are now enabled
LOG:  autovacuum launcher started
LOG:  database system is ready to accept connections
LOG:  invalid length of startup packet
ERROR:  database "greenlight_production" already exists
STATEMENT:  CREATE DATABASE "greenlight_production" ENCODING = 'utf8'
LOG:  received fast shutdown request
LOG:  aborting any active transactions
LOG:  autovacuum launcher shutting down
LOG:  shutting down
LOG:  database system is shut down

PostgreSQL Database directory appears to contain a database; Skipping initialization

LOG:  database system was shut down at 2020-12-19 17:50:38 UTC
LOG:  MultiXact member wraparound protections are now enabled
LOG:  database system is ready to accept connections
LOG:  autovacuum launcher started
LOG:  invalid length of startup packet
ERROR:  database "greenlight_production" already exists
STATEMENT:  CREATE DATABASE "greenlight_production" ENCODING = 'utf8'


--

I need to rest now. I will come back tomorrow.
:-)
Any comment will be appreciated.
Thank you very much for your help.

May you, your family and all beings around you be well, happy and peaceful.

**Comment by Dennis1993** [comment]
See that: https://groups.google.com/g/bigbluebutton-setup/c/F2MlW6Voj-0/m/ZXDq5_-uEQAJ

there is a solution for your Problem. IPv6. Search for „solution“ on the page

**Comment by lovingkindness** [comment]
Hi Dennis1993,
Thank you very much for your help and the link.
:-)
I went through the thread.

1.
Actually I could not understand the issue on this link very well.
My VPS supports both ipv4 and ipv6. So I thought this issue on this link may not be exactly same as what I am encountering.
And also, in my case, ./fs_cli command is not working, gives me an error which I pasted below.
So far, I haven't understood exactly the cause of the problem I am experiencing and what was affected inside my BBB VPS by upgrading to 2.2.31.
If you can see any critical issue in my data, will you point them out? I will be grateful if you could show me the points.
Will you let me know what you think on this?


2.
This is my first upgrading of BBB. My fresh install of 2.2.30 was working gracefully without any problem within my usage.
Is it usual that every time, or often times, upgrading of BBB will cause troubles? By knowing this I will be more careful before upgrading BBB next time.

Are we or am I the only one of the very few people who are affected by the issue I am experiencing? It actually seems no other people is concerned with the issue I am experiencing as long as I keep monitoring other "Issues" posted.


3.
Is it more practical to install BBB freshly on a fresh server environment on every upgrade rather than keep repeating to upgrade BBB on one server to avoid issues caused by every upgrade and at the same time to avoid spending time for troubleshooting of BBB after upgrading?

If so, it seems to me that ideal environment for me is to set up a scalelite and place at least 2 or 3 VPS with it. Then regarding the maintenance and upgrades can be done one machine at a time and even when a machine is troubled by the upgrading, other working machines can still work normally until I fresh install BBB on the troubled machine.


4.
I am still new to running BBB and just started to my test-run of BBB on VPS before purchasing bare-metal machine. So far I am very confident with BBB for my requirements in terms of stability, usability and compatibility with user devices available in the market today. Thank you very much for the community's contribution and generously sharing BBB.


5.
I will place the data collected from my server below as a reference hoping this can be of help of someone also.

Any comment will be appreciated. Thank you.


--

It seems there is no connection to freeswitch is happening since there is no record of 2020-12-20 despite of the fact that I have been testing the server today(Dec 20).
less /opt/freeswitch/log/freeswitch.log

2020-12-19 17:32:28.365374 [CONSOLE] switch_loadable_module.c:1803 Successfully Loaded [mod_logfile]
2020-12-19 17:32:28.366241 [CONSOLE] switch_loadable_module.c:1803 Successfully Loaded [mod_enum]
2020-12-19 17:32:28.366260 [NOTICE] switch_loadable_module.c:292 Adding Dialplan 'enum'
2020-12-19 17:32:28.366284 [NOTICE] switch_loadable_module.c:350 Adding Application 'enum'
2020-12-19 17:32:28.366347 [NOTICE] switch_loadable_module.c:412 Adding API Function 'enum'
2020-12-19 17:32:28.366376 [NOTICE] switch_loadable_module.c:412 Adding API Function 'enum_auto'
2020-12-19 17:32:28.366833 [CONSOLE] switch_loadable_module.c:1803 Successfully Loaded [mod_event_socket]
2020-12-19 17:32:28.366851 [NOTICE] switch_loadable_module.c:350 Adding Application 'socket'
2020-12-19 17:32:28.366873 [NOTICE] switch_loadable_module.c:412 Adding API Function 'event_sink'
2020-12-19 17:32:28.368281 [INFO] mod_sofia.c:6247 Starting initial message thread.
2020-12-19 17:32:28.368494 [DEBUG] sofia.c:4628 debug [0]
2020-12-19 17:32:28.368509 [DEBUG] sofia.c:4628 sip-trace [no]
2020-12-19 17:32:28.368514 [DEBUG] sofia.c:4628 sip-capture [no]
/opt/freeswitch/log/freeswitch.log


--


./fs_cli
[ERROR] fs_cli.c:1691 main() Error Connecting []


--

So I tried this command.

./fs_cli -d 7
[DEBUG] esl_config.c:56 esl_config_open_file() Configuration file is /root/.fs_cli_conf.
[DEBUG] esl_config.c:56 esl_config_open_file() Configuration file is /etc/fs_cli.conf.
[DEBUG] fs_cli.c:1641 main() no profiles found, using builtin profile
[DEBUG] fs_cli.c:1680 main() Using profile ubuntu-8gb-nbg1-1 [127.0.0.1]
[DEBUG] esl.c:1303 esl_recv_event() RECV HEADER [Content-Type] = [auth/request]
[DEBUG] esl.c:1467 esl_recv_event() RECV MESSAGE
Event-Name: SOCKET_DATA
Content-Type: auth/request

[DEBUG] esl.c:1495 esl_send() SEND
auth ClueCon

[DEBUG] esl.c:1303 esl_recv_event() RECV HEADER [Content-Type] = [command/reply]
[DEBUG] esl.c:1303 esl_recv_event() RECV HEADER [Reply-Text] = [-ERR invalid]
[DEBUG] esl.c:1467 esl_recv_event() RECV MESSAGE
Event-Name: SOCKET_DATA
Content-Type: command/reply
Reply-Text: -ERR invalid

[ERROR] fs_cli.c:1691 main() Error Connecting []

Usage: ./fs_cli [-H <host>] [-P <port>] [-p <secret>] [-d <level>] [-x command] [-t <timeout_ms>] [profile]


--


Firefox Console between "Connecting to echo test", and "How would you like to join the audio?" 

Sun Dec 20 2020 22:19:01 | sip.Inviter | Session iao9bvv3vf25mvaonir3hmgiprctcs transitioned to state Establishing sip.js:9602:25
[22:19:01:0222]  INFO: clientLogger: Audio call session progress update https://[HOSTNAME.DOMAIN]/html5client/7ab5ebd28d0d066903f9363ec402d0fee931a588.js?meteor_js_resource=true:121:1203724 7ab5ebd28d0d066903f9363ec402d0fee931a588.js:9:353686
Sun Dec 20 2020 22:19:01 | sip.Inviter | Inviter.onTrying sip.js:9602:25
Sun Dec 20 2020 22:19:01 | sip.Inviter | Inviter.onReject sip.js:9602:25
Sun Dec 20 2020 22:19:01 | sip.Inviter | Session iao9bvv3vf25mvaonir3hmgiprctcs transitioned to state Terminated sip.js:9602:25
[22:19:01:0496] ERROR: clientLogger: Audio call terminated. cause=ICE error https://[HOSTNAME.DOMAIN]/html5client/7ab5ebd28d0d066903f9363ec402d0fee931a588.js?meteor_js_resource=true:121:1205214 7ab5ebd28d0d066903f9363ec402d0fee931a588.js:9:353686
[22:19:01:0499] ERROR: clientLogger: Audio error - errorCode=1004, cause=ICE error https://[HOSTNAME.DOMAIN]/html5client/7ab5ebd28d0d066903f9363ec402d0fee931a588.js?meteor_js_resource=true:121:1160382 7ab5ebd28d0d066903f9363ec402d0fee931a588.js:9:353686
Sun Dec 20 2020 22:19:01 | sip.Inviter | Session iao9bvv3vf25mvaonir3hmgiprctcs in state Terminated is being disposed

--


Firefox about:webrtc


(registry/INFO) insert 'ice' (registry) succeeded: ice

(registry/INFO) insert 'ice.pref' (registry) succeeded: ice.pref

(registry/INFO) insert 'ice.pref.type' (registry) succeeded: ice.pref.type

(registry/INFO) insert 'ice.pref.type.srv_rflx' (UCHAR) succeeded: 0x64

(registry/INFO) insert 'ice.pref.type.peer_rflx' (UCHAR) succeeded: 0x6e

(registry/INFO) insert 'ice.pref.type.host' (UCHAR) succeeded: 0x7e

(registry/INFO) insert 'ice.pref.type.relayed' (UCHAR) succeeded: 0x05

(registry/INFO) insert 'ice.pref.type.srv_rflx_tcp' (UCHAR) succeeded: 0x63

(registry/INFO) insert 'ice.pref.type.peer_rflx_tcp' (UCHAR) succeeded: 0x6d

(registry/INFO) insert 'ice.pref.type.host_tcp' (UCHAR) succeeded: 0x7d

(registry/INFO) insert 'ice.pref.type.relayed_tcp' (UCHAR) succeeded: 0x00

(registry/INFO) insert 'stun' (registry) succeeded: stun

(registry/INFO) insert 'stun.client' (registry) succeeded: stun.client

(registry/INFO) insert 'stun.client.maximum_transmits' (UINT4) succeeded: 7

(registry/INFO) insert 'ice.trickle_grace_period' (UINT4) succeeded: 5000

(registry/INFO) insert 'ice.tcp' (registry) succeeded: ice.tcp

(registry/INFO) insert 'ice.tcp.so_sock_count' (INT4) succeeded: 0

(registry/INFO) insert 'ice.tcp.listen_backlog' (INT4) succeeded: 10

(registry/INFO) insert 'ice.tcp.disable' (char) succeeded: \000

(registry/INFO) insert 'ice.forced_interface_name' (string) succeeded:

(generic/EMERG) Exit UDP socket connected

(ice/INFO) /builds/worker/checkouts/gecko/dom/media/webrtc/transport/third_party/nICEr/src/net/nr_socket_multi_tcp.c:173 function nr_socket_multi_tcp_create_stun_server_socket skipping UDP STUN server(addr:)

(ice/WARNING) /builds/worker/checkouts/gecko/dom/media/webrtc/transport/third_party/nICEr/src/net/nr_socket_multi_tcp.c:617 function nr_socket_multi_tcp_listen failed with error 3

(ice/WARNING) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf)): failed to create passive TCP host candidate: 3

(ice/INFO) /builds/worker/checkouts/gecko/dom/media/webrtc/transport/third_party/nICEr/src/net/nr_socket_multi_tcp.c:173 function nr_socket_multi_tcp_create_stun_server_socket skipping UDP STUN server(addr:)

(ice/WARNING) /builds/worker/checkouts/gecko/dom/media/webrtc/transport/third_party/nICEr/src/net/nr_socket_multi_tcp.c:617 function nr_socket_multi_tcp_listen failed with error 3

(ice/WARNING) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf)): failed to create passive TCP host candidate: 3

(generic/ERR) NrTcpSocket::OnClose 000000001D044000 reason=2152398868 name=NS_ERROR_NET_RESET

(ice/ERR) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf)): Error 10 on reliable socket(000000001D044000). Abandoning.

(generic/ERR) NrTcpSocket::OnClose 00000000148B6C00 reason=2152398868 name=NS_ERROR_NET_RESET

(ice/ERR) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf)): Error 10 on reliable socket(00000000148B6C00). Abandoning.

(generic/ERR) NrTcpSocket::OnClose 000000001D043400 reason=2152398868 name=NS_ERROR_NET_RESET

(ice/ERR) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf)): Error 10 on reliable socket(000000001D043400). Abandoning.

(generic/ERR) NrTcpSocket::OnClose 000000001430F800 reason=2152398868 name=NS_ERROR_NET_RESET

(ice/ERR) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf)): Error 10 on reliable socket(000000001430F800). Abandoning.

(ice/WARNING) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf)): peer (PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf):default) has no stream matching stream PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf) transport-id=transport_0 - 655ce3c7:1413148905ec3aaae4e133d2c56228a6

(ice/WARNING) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf))/CAND(srflx(IP4:10.111.0.66:51842/UDP|[HOSTNAME.DOMAIN]:3478)): failed to initialize, 5 remaining

(ice/WARNING) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf))/CAND(relay(IP4:10.111.0.66:0/TLS|[HOSTNAME.DOMAIN]:443)): failed to initialize, 4 remaining

(ice/WARNING) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf))/CAND(relay(IP4:10.111.0.66:0/TCP|[HOSTNAME.DOMAIN]:443)): failed to initialize, 3 remaining

(ice/WARNING) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf))/CAND(srflx(IP4:10.111.0.66:51843/UDP|[HOSTNAME.DOMAIN]:3478)): failed to initialize, 2 remaining

(ice/WARNING) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf))/CAND(relay(IP4:10.111.0.66:0/TLS|[HOSTNAME.DOMAIN]:443)): failed to initialize, 1 remaining

(ice/WARNING) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf))/CAND(relay(IP4:10.111.0.66:0/TCP|[HOSTNAME.DOMAIN]:443)): failed to initialize, 0 remaining

(ice/INFO) ICE(PC:1608473931480000 (id=15032385589 url=https://[HOSTNAME.DOMAIN]/html5client/join?sessionToken=vkk9figjuznlhabf)): All candidates initialized

+++++++ END ++++++++


Thank you

May you, your family and all beings around you be well, happy and peaceful.

**Comment by farhatahmad** [comment]
Hi , 

This is the repository for Greenlight, the front-end for BigBlueButton. You'd be better off posting this in the BigBlueButton forum instead (https://groups.google.com/forum/#!forum/bigbluebutton-setup)

**Comment by githlp** [comment]
i am not so sure about that - i have no problem starting a bbb room session initiated by moodle.

But i cannot start it from greenlight - its the same error 1006 with my installation when trying to connect for echo-test..

Tried the "ip4 only" parameter in external /opt/freeswitch/etc/freeswitch/sip_profiles/external.xml even though i have ip6 ready going, but it keeps to remain a no-go from the docker-greenlight front-end.

No idea how to fix that..

**Comment by lovingkindness** [comment]
Hello,

@githlp,
Thank you for sharing your experience.

@farhatahmad,
First of all, here is my apology to you and repository members for my misplacing the issue here.

Honestly I really could not tell if this issue was originated with Greenlight or BBB when I posted. I do not have enough knowledge to isolate this issue. When I tried to open a new issue, the given options by the system seem to me were to choose between issues with Greenlight installed environment or not. So I chose Greenlight although I could not be sure if the issue was happening within the Greenlight or not.

As per your advice, I understood that my issue is not related at all to Greenlight. Thank you very much.

2.
It would be great if you could help me with clarification before I move this issue to BigBlueButton repository.

Since I do not have experience joining BBB room without using Greenlight, I can not tell the beginning and end of Greenlight's roll in a user session. Because Greenlight is very well integrated with BBB. May I know which stage of user session is the clean separation point of Greenlight and BBB in one user session of signing into the account and logging out from the account?

A-1. "Connecting to echo test..." window itself is not part of the Greenlight. So this issue is originated with BBB.
A-2. "Connecting to echo test..." window itself is part of the Greenlight. But at the back of the window, Greenlight is trying to establish connection to BBB and BBB is not responding correctly. But Greenlight itself is doing the correct and expected job. So this issue is originated with BBB.

B-1. "Failure on call (Reason=ICE error) (error 1004)" window is not part of the Greenlight. So this issue is originated with BBB.
B-2. "Failure on call (Reason=ICE error) (error 1004)" window is part of the Greenlight. But the information appeared on the window was given from the BBB. Greenlight is just displaying the information given by the BBB. So this issue is originated with BBB.

C Both above A & B are incorrect. Accessing BBB via the front end, Greenlight, Greenlight's function is just to cast whatever information user provided to BBB at the back-end and displays to users the whatever information received from BBB. From the beginning of user's signing in to an account until the user's log out from the account, user stays within the Greenlight during entire session. Greenlight is a wrapper of BBB for users.


I would be grateful if you could help me to understand, how you could isolate this issue from Greenlight? I just want to learn the concept of how to isolate the issues.

Any comment will be appreciated.

Thank you

May you, your family and all beings around you be well, happy and peaceful.

**Comment by githlp** [comment]
Ok, i was able to fix it:

`sudo bbb-conf --setip sub.mydomain.de`
`sudo bbb-conf --clean`
`sudo bbb-conf --check`

works now ootb

**Comment by ffdixon** [comment]
For the benefit of others, these are the steps listed in our documentation for manually upgrading to the latest build of BigBlueButton 2.2.

  https://docs.bigbluebutton.org/2.2/install.html#upgrading-from-bigbluebutton-22

**Comment by githlp** [comment]
Thank you!

Now how exactly do i use the  bbb-install.sh script for plain updating from alpha-2 to alpha-3 ? (i purged the pkgs already following your comment)

In the documentation it explains system updates and very detailed several options of an initial bbb-installation, but i think i have to omit the certificate and email part

`-s bbb.example.com -e info@example.com -w -x `

since that is already in place, so it's just this command:

`wget -qO- https://ubuntu.bigbluebutton.org/bbb-install.sh | bash -s -- -v bionic-230-dev`

**Comment by lovingkindness** [comment]
Hello,

@ffdixon
Thank you very much for your generosity to share BBB with the world and also for your help here by discovering us buried in the "Closed" tagged issues.
Only after my suffering from pains caused by this trouble, I can now understand much better the meanings of what is explained in the document you linked.
:-)


@githlp,
Thank you very much for sharing your experiences and joining in the thread. I hope your issue will be resolved soon.



With regards to my final solution to this issue, I deleted the troubled VPS and did the fresh install of BBB 2.2.31 and it has been working fine.

Below are the links which helped me to reinstall the BBB.

--

If everything above fails, just reinstalling BigBlueButton should be enough, but a fresh ubuntu install won’t hurt if things were messed up.
https://higheredlab.com/bigbluebutton-guide/#bigbluebutton-troubleshooting

How to completely remove BigBlueButton from Ubuntu
https://higheredlab.com/bigbluebutton-guide/#bigbluebutton-remove

--

Happy holidays to all of you.

May you, your family and all beings around you be well, happy and peaceful.
